src/absil/ilwrite.fs

```diff
@@ -4025,17 +4074,18 @@ let writeBinaryAndReportMappings (outfile, ilg, pdbfile: string option, signer:
           let entrypointCodeChunk,next = chunk 0x06 next
           let globalpointerCodeChunk,next = chunk (if isItanium then 0x8 else 0x0) next
           
-          let debugDirectoryChunk,next = chunk (if pdbfile = None then 0x0 else sizeof_IMAGE_DEBUG_DIRECTORY) next
+          let debugDirectoryChunk,next = chunk (if not pdbfile then 0x0 else sizeof_IMAGE_DEBUG_DIRECTORY) next
           // The debug data is given to us by the PDB writer and appears to 
           // typically be the type of the data plus the PDB file name.  We fill 
-          // this in after we've written the binary. We approximate the size according 
+          // this in after we've written the binary. We approximate the size to MAXPATH according 
           // to what PDB writers seem to require and leave extra space just in case... 
           let debugDataJustInCase = 40
+          let MAX_PATH = 260 //windows MAX_PATH, the maximum length of a file path
           let debugDataChunk,next = 
               chunk (align 0x4 (match pdbfile with 
-                                | None -> 0x0 
-                                | Some f -> (24 
-                                            + System.Text.Encoding.Unicode.GetByteCount(f) // See bug 748444
+                                | false -> 0x0 
+                                | true -> (24 
+                                            + MAX_PATH
```

Comment:
I think I still want this to preserve the old behaviour in the case that we have a named file.  I just don't want us to have any chance of a regression here where we do something wrong with long file names.